tests/tests/runner_tests.rs

```diff
@@ -309,6 +309,52 @@ async fn derived_loaders() {
     );
 }
 
+#[tokio::test]
+async fn end_block() -> anyhow::Result<()> {
```

Comment:
is it possible to add a test for the revert to make sure we have coverage of the filter rebuild?

Reply:
Just made the changes for making the filter rebuild testable